pkg/sriov-fec/nodeconfig.go

```diff
@@ -0,0 +1,357 @@
+package sriovfec
+
+import (
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	"github.com/openshift-kni/eco-goinfra/pkg/sriov-fec/sriovfectypes"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	sriovfecnodeconfig = "SriovFecNodeConfig"
+)
+
+// NodeConfigBuilder provides struct for the SriovFecNodeConfig object containing connection to
+// the cluster and the SriovFecNodeConfig definitions.
+type NodeConfigBuilder struct {
+	// SriovFecNodeConfig definition. Used to create SriovFecNodeConfig object.
+	Definition *sriovfectypes.SriovFecNodeConfig
+	// Create SriovFecNodeConfig object.
+	Object *sriovfectypes.SriovFecNodeConfig
+	// apiClient opens a connection to the cluster.
+	apiClient *clients.Settings
+	// Used in functions that define SriovFecNodeConfig definitions. errorMsg is processed before SriovFecNodeConfig
+	// object is created.
+	errorMsg string
+}
+
+// AdditionalOptions additional options for sriovfecnodeconfig object.
+type AdditionalOptions func(builder *NodeConfigBuilder) (*NodeConfigBuilder, error)
+
+// NewNodeConfigBuilder creates a new instance of NodeConfigBuilder.
+func NewNodeConfigBuilder(
+	apiClient *clients.Settings,
+	name, nsname string,
+	label map[string]string,
+) *NodeConfigBuilder {
```

Comment:
Please format function according to code convention.
func NewNodeConfigBuilder(
	apiClient *clients.Settings,
	name, nsname string,
	label map[string]string) *NodeConfigBuilder {

Reply:
Thanks, updated.